### PR TITLE
Enable full DNS names

### PR DESCRIPTION
**Pull request by jakub-racek-swi**
**Description:**
Expanding the field for the collection of DNS names and by extension workload names in metric labels. The names of workloads can get long, for example if there is a certain service running on a certain pod in a certain namespace all that information gets concatenated into this string. It was limited to 80 characters before, while 256 characters is the actual theoretical limit.

This branch was migrated from https://github.com/open-telemetry/opentelemetry-network/pull/260

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-network/issues/256

**Testing:** Manual testing, running in devbox

**Documentation:** none

### CODE REVIEW

**Review by yonch — APPROVED** [review]
lgtm. Thank you!

**Comment by jakub-racek-swi** [comment]
Just leaving it here that I cannot merge the PR myself as I do not have write access 😄

**Pull request merged**
